graders/c/cgrader/cgrader.py

```diff
@@ -238,7 +263,48 @@ def add_test_result(self, name, description='', points=True,
                 self.result['partial_scores'][field]['points'] += points
                 self.result['partial_scores'][field]['max_points'] += max_points
         return test
-                
+
+    def run_check_suite(self, exec_file, args=None,
+                        use_suite_title=False, use_case_name=True, use_unit_test_id=True,
+                        use_iteration=False):
+
+        if not args:
+            args = []
+        if not isinstance(args, list):
+            args = [args]
+
+        log_file_dir = tempfile.mkdtemp()
+        log_file = os.path.join(log_file_dir, 'tests.xml')
+        self.change_mode(log_file_dir, '777', change_parent=False)
+    
+        out = self.run_command([exec_file] + args,
+                               env={'CK_XML_LOG_FILE_NAME': log_file, 'TEMP': '/tmp'})
+        print(out)
+
+        separator_1 = ': ' if use_suite_title and use_case_name else ''
+        separator_2 = ' - ' if use_unit_test_id and (use_suite_title or use_case_name) else ''
+        try:
+            with open(log_file) as f:
+                print('DEBUG: LOG FILE')
+                print(f.read())
+
+            tree = ET.parse(log_file)
+            for suite in tree.getroot().findall('{*}suite'):
+                suite_title = suite.findtext('{*}title') if use_suite_title else ''
+                for test in suite.findall('{*}test'):
+                    result = test.get('result')
+                    test_id = test.findtext('{*}id') if use_unit_test_id else ''
+                    iteration = f" (run {test.findtext('{*}iteration')})" if use_iteration else ''
+                    case_name = test.findtext('{*}description') if use_case_name else ''
+                    self.add_test_result(f'{suite_title}{separator_1}{case_name}{separator_2}{test_id}{iteration}',
+                                         points=result == 'success',
```

Comment:
Food for thought: in the Catch-based autograder we developed for CS 225 at UIUC, we could use tags to specify additional metadata or execution instructions for individual tests, such the number of points a given test case is worth, or whether or not a given test case should be executed with Valgrind. Looks like recent version of Check include `tcase_set_tags`, which could also be used for this! Unfortunately, per the versions chart at https://github.com/libcheck/check#packaging, the version of Check published for Ubuntu 18.04 isn't recent enough to include that 😢

Reply:
I was considering using this opportunity to upgrade to 20.04, but that is not far enough either. And I'd prefer to avoid non-LTS versions... I'll figure some things out.

Reply:
I was thinking the same things. I'm be no means asking for a change here - just throwing out an idea that may someday be useful!

Reply:
Added as issue #5126 .